test/test_model.py

```diff
@@ -12,86 +12,46 @@
-            'db2': [5, 6],
-        }[relation_name]
-
-    def relation_list(self, relation_id):
-        try:
-            return {
-                4: ['remoteapp1/0'],
-                5: ['remoteapp1/0'],
-                6: ['remoteapp2/0'],
-            }[relation_id]
-        except KeyError:
-            raise op.model.RelationNotFoundError()
-
-    def relation_get(self, relation_id, member_name):
-        try:
-            return {
-                4: {
-                    'myapp/0': {'host': 'myapp-0'},
-                    'remoteapp1/0': {'host': 'remoteapp1-0'},
-                },
-                5: {
-                    'myapp/0': {'host': 'myapp-0'},
-                    'remoteapp1/0': {'host': 'remoteapp1-0'},
-                },
-                6: {
-                    'myapp/0': {'host': 'myapp-0'},
-                    'remoteapp2/0': {'host': 'remoteapp2-0'},
-                },
-            }[relation_id][member_name]
-        except KeyError:
-            raise op.model.RelationNotFoundError()
-
-    def relation_set(self, relation_id, key, value):
-        if relation_id == 5:
-            raise ValueError()
-        self.relation_set_calls.append((relation_id, key, value))
-
-    def config_get(self):
-        return {
-            'foo': 'foo',
-            'bar': 1,
-            'qux': True,
-        }
-
 class TestModel(unittest.TestCase):
 
     def setUp(self):
-        self.backend = FakeModelBackend()
+        os.environ['JUJU_UNIT_NAME'] = 'myapp/0'
+        self.addCleanup(os.environ.pop, 'JUJU_UNIT_NAME')
+
+        self.backend = op.model.ModelBackend()
         meta = op.charm.CharmMeta()
         meta.relations = {'db0': None, 'db1': None, 'db2': None}
         self.model = op.model.Model('myapp/0', meta, self.backend)
 
-        os.environ['JUJU_UNIT_NAME'] = 'myapp/0'
-        self.addCleanup(os.environ.pop, 'JUJU_UNIT_NAME')
-
     def test_model(self):
         self.assertIs(self.model.app, self.model.unit.app)
 
     def test_relations_keys(self):
+        fake_script(self, 'relation-ids',
+                    """[ "$1" = db2 ] && echo '["db2:5", "db2:6"]' || echo '[]'""")
+        fake_script(self, 'relation-list',
+                    """([ "$2" = 5 ] && echo '["remoteapp1/0", "remoteapp1/1"]') || ([ "$2" = 6 ] && echo '["remoteapp2/0"]') || exit 2""")
+
         for relation in self.model.relations['db2']:
             self.assertIn(self.model.unit, relation.data)
             unit_from_rel = next(filter(lambda u: u.name == 'myapp/0', relation.data.keys()))
             self.assertIs(self.model.unit, unit_from_rel)
 
+        self.assertEqual(fake_script_calls(self), [
+            ['relation-ids', 'db2', '--format=json'],
+            ['relation-list', '-r', '5', '--format=json'],
+            ['relation-list', '-r', '6', '--format=json']])
+
     def test_get_relation(self):
+        RELATION_ERROR_MESSAGE = "ERROR invalid value \"$2\" for option -r: relation not found"
```

Comment:
This will look better as a usual local variable, using single quotes as usual so we don't need to escape the quotes inside it:
```
err_msg = '..."..."...'`
```